### PR TITLE
Code has been modularized a bit and now uses numpy for calculations

### PR DESCRIPTION
**Pull request by ejgal**
Some improvements to the code.

### CODE REVIEW

File: src/improvement.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+for num_measurements in [25, 100, 500]:
+
+   temperatures = read_data(
+       file_name="data/temperatures.csv",
+       nrows=num_measurements,
+       column="Air temperature (degC)",
+   )
+
+   mean = np.mean(temperatures)
```
Comment (huitang-earth) [inline src/improvement.py]: Can you add a comment to explain the fuction. :)

**Review by huitang-earth — CHANGES_REQUESTED** [review]
It looks great!

**Comment by asmundb** [comment]
Thanks for your effort, I think it's a good idea to use the numpy library in this project.

One question: have you considered the possibility for nan values in the dataset? One could perhaps want to use the numpy.nanmean instead.

File: src/improvement.py (hunk 1 above)
Comment (ejgal) [reply src/improvement.py]:
```suggestion
   # This calculates the arithmetic mean of the temperatures
   mean = np.mean(temperatures)
```

**Comment by ejgal** [comment]
Good point @asmundb. I'll change this.

File: src/improvement.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+   plt.xlabel(xlabel)
+   plt.ylabel(ylabel)
+   plt.axhline(y=mean, color="b", linestyle="--")
+   plt.savefig(file_name)
+   plt.clf()
+
+
+def read_data(file_name, nrows, column):
+   data = pd.read_csv(file_name, nrows=nrows)
+   return data[column]
```
Comment (ejgal) [inline src/improvement.py:16]:
```suggestion
def read_DATA(file_name, nrows, column):
   data = pd.read_csv(file_name, nrows=nrows)
   return data[column]
```

**Review by huitang-earth — APPROVED** [review]

File: src/improvement.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+for num_measurements in [25, 100, 500]:
+
+   temperatures = read_data(
+       file_name="data/temperatures.csv",
+       nrows=num_measurements,
+       column="Air temperature (degC)",
+   )
+
+   # This calculates the arithmetic mean of the temperatures
+   mean = np.nanmean(temperatures)
```
Comment (asmundb) [inline src/improvement.py:27]:
```suggestion
   mean = np.nanmean(temperatures)
   if np.nan in temperatures:
       print("WARNING: dataset contain nans, mean is based on non nan values")
```

Comment (asmundb) [reply src/improvement.py:27]: I suggest a warning here, perhaps it is reasonable to introduce a warning [warning module ](https://docs.python.org/3/library/warnings.html#module-warnings)

**Review by asmundb — CHANGES_REQUESTED** [review]
see suggested modifications about warning the user about nan values

**Pull request closed without merging**